onvif/src/discovery/mod.rs

```diff
@@ -115,9 +117,20 @@ pub async fn discover(duration: Duration) -> Result<impl Stream<Item = Device>,
     let xml_receiver = ReceiverStream::new(xml_receiver);
     let device_receiver = ReceiverStream::new(device_receiver);
 
+    let mut known_responses = HashSet::new();
+
     let produce_xmls = async move {
-        while let Ok(xml) = recv_string(&socket).await {
-            debug!("Probe match XML: {}", xml);
+        while let Ok((xml, src)) = recv_string(&socket).await {
+            if !known_responses.insert(calculate_hash(&xml)) {
```

Comment:
It'll be strange to have duplicates from different sources because XML contains IP addresses of the camera. Hashing `(xml, src)` will give the same amount of collisions, from what I can tell.